File: sieve_of_erastothenes/sieve_of_erastothenes.cpp
#include<iostream>
#include<stdio.h>

using namespace std;

int main()
  {
       int p[10000],n;
       cout<<"Enter N:";
       cin>>n;
      p[2]= 1;
        for(int k=2;k<=n;k++)
          {
               p[k]=1;
          }
        for(int i=2;i*i<=n;i++)
             {
                  if(p[i]==1)
                    {
                          for(int j=2*i;j<=n;j+=i)
                             p[j]=0;
                    }
            }
      cout<<"The prime numbers upto n are :";
       for(int i=2;i<=n;i++)
          {
                if(p[i]==1)
                  cout<<i<<" ";
          }
 return 0;
 }

File: Sieve_of_Erastothenes/sieve_of_erastothenes.py
def SieveOfEratosthenes(n):
    prime = [True for i in range(n+1)]
    p = 2
    while (p * p <= n):
        if (prime[p] == True):
            for i in range(p * 2, n+1, p):
                prime[i] = False
        p +=1
        for p in range(2, n):
            if prime[p]:
                print p,

# driver program
def main():
    n = input("Enter the limitting number")
    print "Following are the prime numbers smaller",
    print "than or equal to", n
    SieveOfEratosthenes(n)

main()
